### PR TITLE
Plugins: Fix Cache-Control header response for public/plugins/* assets API

### PR DESCRIPTION
**Pull request by wbrowne**
**What is this feature?**

Adds the correct Cache-Control header response for public plugin assets

**Why do we need this feature?**

All assets served from `/public/plugins/{pluginID}/*` have the `Cache-Control: no-store`  header set, which is overwriting the headers defined in the plugins asset handler

**Special notes for your reviewer**:
Thanks to @going-confetti for spotting this and providing all the necessary information to debug and fix ❤️ 

CC @kylebrandt as you have been one to touch some of this recently - let me know if this is the intended work around for allowing caching overrides

**NOTE**: This also adds new behaviour to permit public Cache Control

### CODE REVIEW

**Review by andresmgot — APPROVED** [review]

**Review by academo — APPROVED** [review]

File: pkg/api/plugins.go [hunk 1]
```diff
@@ -366,9 +366,11 @@ func (hs *HTTPServer) serveLocalPluginAsset(c *contextmodel.ReqContext, plugin p
 	}
 
 	if hs.Cfg.Env == setting.Dev {
-		c.Resp.Header().Set("Cache-Control", "max-age=0, must-revalidate, no-cache")
+		c.Resp.Header().Set("Cache-Control", "private, max-age=0, must-revalidate, no-cache")
+		c.Resp.Header().Set("X-Grafana-Cache", "true")
 	} else {
-		c.Resp.Header().Set("Cache-Control", "public, max-age=3600")
```
Comment (wbrowne) [inline pkg/api/plugins.go:371]: @kylebrandt is there a way to introduce these overrides without having to add `private`?

**Review by xnyo — APPROVED** [review]

File: pkg/api/plugins.go (hunk 1 above)
Comment (kylebrandt) [reply pkg/api/plugins.go:371]: Not currently. We would want to make sure it is intentional. Maybe another key to X-Grafana-Cache?

Comment (wbrowne) [reply pkg/api/plugins.go:371]: Alrighty - just did a quick and dirty change.

**Comment by marefr** [comment]
Possibly not applicable comment to these changes. Think we've discussed earlier to fetch plugin assets with the version included in the route. If doing this it could allow to cache the assets **"forever"**. Is this something we should prioritize moving forward?

**Comment by wbrowne** [comment]
Yeah not a bad idea @marefr - I wonder could this open some new challenges if we have to deal with versions that wind up with problems (latest worldmap for example) - curious what @grafana/plugins-platform-frontend think about it

**Comment by academo** [comment]
@wbrowne considering we often (or never) re-release plugin versions having a permanent cache is not such a bad idea. But (to me) still sounds far-reaching, we can for instance set a weeks or months cache to be in the safe side.

**Review by marefr — APPROVED** [review]
💥

**Pull request merged**